src/interface/src/app/signup/signup.component.ts

```diff
@@ -12,6 +12,8 @@ export class SignupComponent implements OnInit {
 
   error: any;
 
+  model: any = {};
```

Comment:
Can we rename or type "model" to be more descriptive? Or alternatively we can use formControl, which can add validations.